### ISSUE TITLE
[Bug] `yarn set version classic` fails with `Error: Invalid semver version`

### ISSUE THREAD

**Issue by n-rodriguez** [yarnpkg/berry#2321]
**Describe the bug**

```sh
nicolas@laptop:~/PROJECTS/CONCERTO/concerto$ yarn set version classic
➤ YN0000: Downloading https://nightly.yarnpkg.com/latest.js
➤ YN0001: Error: Invalid semver version
    at /home/nicolas/PROJECTS/CONCERTO/concerto/.yarn/releases/yarn-2.4.0.cjs:2:86456
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async a.mktempPromise (/home/nicolas/PROJECTS/CONCERTO/concerto/.yarn/releases/yarn-2.4.0.cjs:2:513898)
    at async Qe (/home/nicolas/PROJECTS/CONCERTO/concerto/.yarn/releases/yarn-2.4.0.cjs:2:86181)
    at async /home/nicolas/PROJECTS/CONCERTO/concerto/.yarn/releases/yarn-2.4.0.cjs:2:86042
    at async Function.start (/home/nicolas/PROJECTS/CONCERTO/concerto/.yarn/releases/yarn-2.4.0.cjs:2:388437)
    at async we.execute (/home/nicolas/PROJECTS/CONCERTO/concerto/.yarn/releases/yarn-2.4.0.cjs:2:85865)
    at async we.validateAndExecute (/home/nicolas/PROJECTS/CONCERTO/concerto/.yarn/releases/yarn-2.4.0.cjs:2:659490)
    at async Y.run (/home/nicolas/PROJECTS/CONCERTO/concerto/.yarn/releases/yarn-2.4.0.cjs:17:3854)
    at async Y.runExit (/home/nicolas/PROJECTS/CONCERTO/concerto/.yarn/releases/yarn-2.4.0.cjs:17:4021)
➤ YN0000: Failed with errors in 0s 386ms
```

**To Reproduce**

<details>
  <summary>Reproduction</summary>

  ```js repro
yarn set version classic
  ```
</details>

**Comment by n-rodriguez** [comment]
It's the same than https://github.com/yarnpkg/berry/issues/1893

https://nightly.yarnpkg.com/latest.js resolves to https://nightly.yarnpkg.com/yarn-1.23.0-20210102.0010.js which contains leading zeros

**Comment by mhassan1** [comment]
This seems to be caused by the generation of the version [here](https://github.com/yarnpkg/yarn/blob/14bcd15a401e063c2130ab31568c80aeb914165f/scripts/set-dev-version.js#L16):

```js
const formattedDate =
  date.getUTCFullYear() +
  leftPad(date.getUTCMonth() + 1) +
  leftPad(date.getUTCDate()) +
  '.' +
  leftPad(date.getUTCHours()) +
  leftPad(date.getUTCMinutes());
```

Is there any appetite for changing this so that it does not generate invalid semver? Perhaps using a `-` instead of a `.` before the time? Or prepending a non-zero before the hour? I don't know if there's a dependency on the current format.

In our case, this causes breakage to a Yarn v2 workflow that packs a GitHub dependency using Yarn Classic as part of the project installation.

**Comment by n-rodriguez** [comment]
Why not skipping hours and minutes and replace it with git ref :

`yarn-1.23.0-20210102.git-123456.js`

**Comment by mhassan1** [comment]
I've just opened https://github.com/yarnpkg/yarn/issues/8534.

**Comment by merceyz** [comment]
Closing as fixed in https://github.com/yarnpkg/yarn/commit/3119382885ea373d3c13d6a846de743eca8c914b

**Comment by sujitykulkarni** [comment]
This issue has appeared today on my Azure devops pipeline where I try to set yarn version `yarn set version berry`
I wonder why this has shown up today, as there was no such error till yesterday on my build pipelines. What could I be missing here?

**Comment by merceyz** [comment]
@sujitykulkarni This issue was about `yarn set version classic` so your issue is unrelated, please open a new issue with some more details. We released Yarn 3.0 so that might have messed with your CI if you have some incompatible plugins.

**Comment by pravi** [comment]
@sujitykulkarni I noticed this error today as well. If you are on node 10.x and trying to upgrade to yarn 3.x, this will fail as yarn 3.0 is not supported on node 10.x
